### ISSUE TITLE
simplify generate/update process

### ISSUE THREAD

**Issue by kylemcdonald** [openframeworks/projectGenerator#40]
i'd like to propose two changes:

1. every time the user clicks "generate" or "update" we open the folder. this acts as confirmation that it worked. only pop up the modal if there is an error.
2. next to "generate" or "update" we have "generate and launch" or "update and launch" which opens the project file itself.

i think the way it's designed now is very useful for PG development/debugging, but the more i use it the more it feels unnecessary.

**Comment by ofZach** [comment]
I think maybe the modal could just be an option like "show console", for debugging it's *really* valuable, otherwise the behavior is sort of a black box and I don't think we will be able to see things like mangled paths, etc if people have issues.  Also verbose output gives you some insight into what the PG is doing, like if addons don't work.   I'm fine to change the default behavior to opening the folder.   I don't know about adding a second button but it's easy enough to try...

**Comment by Daandelange** [comment]
+1 for the modal too.

Systematically opening the folder it isn't good I think. I usually already have the project folder open in the finder, but with sub-folder views so it would open a new window and pollute windows-space. Plus there's gonna be a drag & drop feature which would act in the same way.

Kyle, when you say Launch, you mean open in IDE, show files, or run the ofApp ?

What about showing show a `Run` button when we detect a compiled binary within the project folder?

**Comment by Daandelange** [comment]
Going from `generate` to `update`, we can show a `launch` button to inform that the command succeeded, but hitting update won't notify anything as the button would already be visible. In this case we could show an ui modal that automatically disappears.  
Another solution could be to change the text of the "generate/update" button to "Success!" for a few seconds, then turn it back to it's normal state.

**Comment by Daandelange** [comment]
This commit adds that 2nd solution: https://github.com/Daandelange/projectGenerator/commit/f348f038f59f52061ce1525c1d3bb3b750485f63

**Comment by kylemcdonald** [comment]
@Daandelange i meant that "launch" should open the project file in the IDE.

i agree that opening another finder/browser window is not very useful.

**Comment by Daandelange** [comment]
I've added the Launch button for XCode to this PR: https://github.com/openframeworks/projectGenerator/pull/55

Other platforms should not be hard to implement and the button needs to be hidden on platforms where this feature is not available.

**Comment by ofZach** [comment]
I've added an "open in IDE" button on the modal which seems to help for workflow (I took most of @Daandelange's PR but the launch button seemed a little weird in the UI, so we use the modal).   I'm happy to revisit the modal / flow conversation later after 0.9...

**Comment by kylemcdonald** [comment]
> Plus there's gonna be a drag & drop feature which would act in the same way.

just want to put another vote in for this :)

**Comment by ofZach** [comment]
is the vote for drag and drop ?  (to pull up update?)

**Comment by kylemcdonald** [comment]
yes, that's a vote for drag and drop :) i was just using the PG heavily to update a bunch of repos, testing that it worked for all of them before removing the project files. good news: it does! bad news: no drag and drop meant that it took a little longer (i also usually work with the finder open, like @Daandelange describes).

**Comment by ofZach** [comment]
drag and drop should come in the next release along with progress indicators for recursive update (which can take a while)...   Afaik most of the code is there already, but there are some UI things to figure out and some interaction things with if you drag a folder vs a project file, etc....

**Comment by Daandelange** [comment]
The drag'n'drop feature is partially done in the commit above.. (needs to be implemented for batch updating, but the code is prepared)

How about the "Launch" feature ? Can we consider this fixed with the "Open in IDE" button in the success modal ?

**Comment by Daandelange** [comment]
@ofZach @kylemcdonald Can we close this issue?
